aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3ObjectMapper.java

```diff
@@ -48,13 +48,16 @@ public class S3ObjectMapper {
 
   private S3ObjectMapper() {}
 
-  static ObjectMapper mapper() {
+  public static ObjectMapper mapper() {
     if (!isInitialized) {
       synchronized (S3ObjectMapper.class) {
```

Comment:
Something I missed earlier but it's minor, generally for these kinds of initializations I think it's more clear to have an AtomicReference<ObjectMapper>() and use the `compareAndSet` method if the instance is null.

Reply:
that is actually a good idea and I agree that we can improve this area. I opened https://github.com/apache/iceberg/pull/6857 to address that